src/utils/daos_hdlr.c

```diff
@@ -2067,40 +2070,39 @@ file_lstat(struct cmd_args_s *ap, struct file_dfs *file_dfs,
 		if (rc != 0)
 			rc = errno;
 	} else if (file_dfs->type == DAOS) {
-		rc = dfs_sys_stat(file_dfs->dfs_sys, path, 0, buf);
+		rc = dfs_sys_stat(file_dfs->dfs_sys, path, O_NOFOLLOW, buf);
 	} else {
+		rc = EINVAL;
 		DH_PERROR_SYS(ap, rc, "File type not known '%s' type=%d", path, file_dfs->type);
 	}
 	return rc;
 }
 
-static ssize_t
+static int
 file_read(struct cmd_args_s *ap, struct file_dfs *file_dfs,
-	  const char *file, void *buf, size_t size)
+	  const char *file, void *buf, daos_size_t *size)
 {
-	ssize_t	    got_size = 0;
 	int	    rc = 0;
 	daos_size_t read_size;
 
 	if (file_dfs->type == POSIX) {
-		got_size = read(file_dfs->fd, buf, size);
+		read_size = read(file_dfs->fd, buf, *size);
+		*size = read_size;
 	} else if (file_dfs->type == DAOS) {
-		read_size = size;
 		rc = dfs_sys_read(file_dfs->dfs_sys, file_dfs->obj, buf, file_dfs->offset,
-				  &read_size, NULL);
+				  size, NULL);
 		if (rc == 0)
 			/* update file pointer with number of bytes read */
-			file_dfs->offset += (daos_off_t)got_size;
-		got_size = read_size;
+			file_dfs->offset += (daos_off_t)*size;
 	} else {
 		DH_PERROR_SYS(ap, rc, "File type not known '%s' type=%d", file, file_dfs->type);
```

Comment:
This still needs to set rc = EINVAL, right?

Reply:
@daltonbohning ah I though I caught all of those again, thanks